activerecord/lib/active_record/associations/association.rb

```diff
@@ -193,9 +193,25 @@ def creation_attributes
           attributes
         end
 
+        def instantiation_attributes(record)
+          attributes = {}
+
+          owners_name = owner.class.base_class.name.downcase
```

Comment:
> Assuming the association name matches the class name seems quite untenable, even if we use the correct transformation (underscore instead of downcase).

@matthewd I think the best way of guaranteeing that the correct association name is found, is by getting it from `Arel::Attribute` . I definitely think this is not a great solution but I could not find a clean way to get the correct association name...

What do you think, is this patch worth merging in or should I close it?